### ISSUE TITLE
Does evaluation on the last batch have ignored the padded data?

### ISSUE THREAD

**Issue by zlmzju** [apache/incubator-mxnet#3488]
I am wondering how to exactly evaluate the testing images using python code.

Following `example/image-classification`, I use python code to train and evaluate the model. I find that `mx.model` does not explicitly handle the padded data of the last batch for validation or testing data.

In [`mx.model.predict()`](https://github.com/dmlc/mxnet/blob/master/python/mxnet/model.py#L613), the padded data have been removed from the outputs, which is what I expected.

I have tried some methods to evaluate the testing datset:
1. Using a proper testing batch size, but it's not feasible when the test set size is a large prime number. And I got an error of `Slice end index out of range` when `testing-batch-size < training-batch-size`, related to this issue #1978.
2. Write my own code to call the `mx.model.predict()` after each epoch to evaluate the validation set.

Please tell me if I am wrong, and hope mxnet has handled this problem for exact evaluation.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
